File: LeetCode/container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0, j = height.size()-1;
        int maxArea = -1;
        
        while(i < j) {
            int area = min(height[i], height[j]) * (j - i);
            if(maxArea < area) {
                maxArea = area;
            }
            if(height[i] < height[j])
                i++;
            else
                j--;
        }
        
        return maxArea;
    }
};

File: LeetCode/container_with_most_water.go
func min(a, b int) int {
    if a < b {
        return a
    }

    return b
}

func max(a, b int) int {
    if a < b {
        return b
    }

    return a
}

func maxArea(height []int) int {
    ans, l, r := 0, 0, len(height)-1

    for l < r {
        area := min(height[l], height[r]) * (r-l)
        ans = max(ans, area)

        if height[l] < height[r] {
            l++
        } else {
            r--
        }
    } 

    return ans
}
